recipes/epics-base/build.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+install -d $PREFIX/bin
+install -d $PREFIX/lib
+install -d $PREFIX/epics
+
+# export EPICS_HOST_ARCH as determined
+EPICS_HOST_ARCH=$(perl src/tools/EpicsHostArch.pl)
+export EPICS_HOST_ARCH
+
+# EPICS assumes gcc suit has /usr/bin prefix.
+# If non-system gcc is used, e.g. rh-devtoolset,
+# find out its location /opt/rh/devtoolset-2/root/usr/bin.
+# GNG_DIR has "bin" stripped.
+GNU_DIR=$(dirname $(dirname $(which gcc)))
+if [ "$GNU_DIR" != "/usr" ]; then
+    echo "GNU_DIR="$GNU_DIR >> configure/CONFIG_COMMON
+fi
+
+make -j$(getconf _NPROCESSORS_ONLN)
+
+EPICS_BASE=$PREFIX/epics
+
+# Copy libraries into $PREFIX/lib
+cp -av $PREFIX/epics/lib/$EPICS_HOST_ARCH/lib*so* $PREFIX/lib 2>/dev/null || : # linux
+cp -av $PREFIX/epics/lib/$EPICS_HOST_ARCH/lib*dylib* $PREFIX/lib 2>/dev/null || :  # osx
```

Comment:
@xiaoqiangwang, do you think we can use `${SHLIB_EXT} here to simplify this to a common line for both unix types?

```bash
# Copy libraries into $PREFIX/lib
cp -av $PREFIX/epics/lib/$EPICS_HOST_ARCH/lib*${SHLIB_EXT} $PREFIX/lib 2>/dev/null || :
```

Also, should we consider using `ln` instead of `cp` here?

Finally, can we consider doing the same for the binaries so that they end up in `${PREFIX}/bin/`?

Reply:
Yes. Using ${SHLIB_EXT} is better. Either copying or linking of libraries/binaries will serve.

Reply:
@xiaoqiangwang, forgive my possible ignorance of how epics works, but does it make sense to set `INSTALL_LOCATION=$(PREFIX)` in the `configure/CONFIG_SITE` so that libraries etc get installed directly into the standard hierarchy? (as opposed to using `$(PREFIX)/epics` and linking things back?)

The only necessary link would then be to soft link from `${PREFIX}/bin/${EPICS_HOST_ARCH}` down to `${PREFIX}/bin/`, I think. Or, we could just patch that location in the config file(s) as well.

Reply:
It is not enough to change ``INSTALL_LOCATION``, but also ``INSTALL_HOST_BIN`` and ``INSTALL_HOST_LIB`` in configure/CONFIG_COMMON. But most epics clients code assume the default layout of the headers, libraries and binaries. This will likely break them,

On another thought, it may not be necessary to link/copy the binaries/libraries. One could use the ``$PREFIX/etc/conda/activate.d/epics_base.sh`` to put the ``$PREFIX/epics/bin/$EPICS_HOST_ARCH`` into ``PATH``. But this requires test.

Reply:
thanks @xiaoqiangwang, can you point me at an example of an epics client that might break. I just want to see what the typical usage is before I actually test breaking everything.

Reply:
One example is my cachannel python package 
https://github.com/CaChannel/CaChannel/blob/master/setup.py#L99. It assumes finding libraries under $EPICS_BASE/lib/$EPICS_HOST_ARCH.

"breaking everything" might be exaggerated. I am not aware of other institutes using the epics-base package in anaconda.

Reply:
This is mainly just me being overly pedantic, but I have a couple of minor changes that I think will help, I'll ping you when their ready for review.

Reply:
I see the point of using standard locations /usr/bin, /usr/lib. But it is also standard to put large libraries like qt5 under /usr/lib64/qt5, and create links to /usr/bin, or add to the PATH.